wdtk-datamodel/src/main/java/org/wikidata/wdtk/datamodel/implementation/MediaInfoDocumentImpl.java

```diff
@@ -0,0 +1,145 @@
+package org.wikidata.wdtk.datamodel.implementation;
+
+/*
+ * #%L
+ * Wikidata Toolkit Data Model
+ * %%
+ * Copyright (C) 2014 Wikidata Toolkit Developers
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+import com.fasterxml.jackson.annotation.*;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import org.wikidata.wdtk.datamodel.helpers.Equality;
+import org.wikidata.wdtk.datamodel.helpers.Hash;
+import org.wikidata.wdtk.datamodel.helpers.ToString;
+import org.wikidata.wdtk.datamodel.interfaces.*;
+
+import java.util.*;
+
+
+/**
+ * Jackson implementation of {@link MediaInfoDocument}.
+ *
+ * @author Thomas Pellissier Tanon
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class MediaInfoDocumentImpl extends LabeledStatementDocumentImpl implements MediaInfoDocument {
+
+	/**
+	 * Constructor.
+	 *
+	 * @param id
+	 *            the id of the media that data is about
+	 * @param labels
+	 *            the list of captions of this media, with at most one label for
+	 *            each language code
+	 * @param statements
+	 *            the list of statement groups of this media info; all of them must
+	 *            have the given itemIdValue as their subject
+	 * @param revisionId
+	 *            the revision ID or 0 if not known; see
+	 *            {@link EntityDocument#getRevisionId()}
+	 */
+	public MediaInfoDocumentImpl(
+			MediaInfoIdValue id,
+			List<MonolingualTextValue> labels,
+			List<StatementGroup> statements,
+			long revisionId) {
+		super(id, labels, statements, revisionId);
+	}
+
+	/**
+	 * Constructor. Creates an object that can be populated during JSON
+	 * deserialization. Should only be used by Jackson for this very purpose.
+	 */
+	@JsonCreator
+	public MediaInfoDocumentImpl(
+			@JsonProperty("id") String jsonId,
+			@JsonProperty("labels") @JsonDeserialize(contentAs=TermImpl.class) Map<String, MonolingualTextValue> labels,
+			@JsonProperty("statements") Map<String, List<StatementImpl.PreStatement>> claims,
```

Comment:
Magnus has complained about this `statements` parameter not being consistent with the `claims` parameter in other entity types… it could potentially be changed to `claims` in the future, maybe? I wonder if it would make sense to support both in this implementation.

Reply:
Indeed, it make sense if the JSON output is changed or if people try to deserialize some custom-made JSON and uses the "claims" field without taking care.